src/gui/settings/qgssettingsregistrygui.h

```diff
@@ -47,6 +47,10 @@ class GUI_EXPORT QgsSettingsRegistryGui : public QgsSettingsRegistry
 #ifndef SIP_RUN
     //! Settings entry respect screen dpi
     static const inline QgsSettingsEntryBool settingsRespectScreenDPI = QgsSettingsEntryBool( QStringLiteral( "respect_screen_dpi" ), QgsSettings::Prefix::GUI_QGIS, false );
+
+    //! Check for plugin updates automatically on startup
+    static const inline QgsSettingsEntryBool settingsAutomaticallyCheckForPluginUpdates = QgsSettingsEntryBool( QStringLiteral( "automatically-check-for-updates" ), QgsSettings::Prefix::PLUGINS, true, QStringLiteral( "Automatically check for plugin updates on startup" ) );
```

Comment:
For the record, to archive a Q & A between myself and the PR author :wink: 

@nirvn: Why putting the settings key in the src/gui registry instead of src/app ?

@nyalldawson : Because app registry isn't available from python.

--

Acceptable compromise it is then.